Modules/xASL_module_ASL.m

```diff
@@ -150,16 +150,45 @@
 
 %% D4. TimeEncoded parsing
 % Check if TimeEncoded is defined
-if isfield(x.Q,'TimeEncodedMatrixType') || isfield(x.Q,'TimeEncodedMatrixSize')
+if isfield(x.Q, 'TimeEncodedMatrixType') || isfield(x.Q, 'TimeEncodedMatrixSize') || isfield(x.Q, 'TimeEncodedMatrix')
     fprintf(2,'Time encoded data detected, we will process this but this is a new feature that is still under development\n');
     x.modules.asl.bTimeEncoded = true;
 	x.modules.asl.bMultiPLD = true;
 	
-	if isempty(x.Q.TimeEncodedMatrixType)
-		fprintf(2,'TimeEncodedMatrixType field missing (should be a Hadamard or Walsh)...\n');
-	elseif ~isfield(x.Q,'TimeEncodedMatrixSize') || isempty(x.Q.TimeEncodedMatrixSize) || (x.Q.TimeEncodedMatrixSize < 2)
-		fprintf(2,'TimeEncodedMatrixSize field missing (should be a multiple of 4)...\n');
-    end
+	% Initialize as empty if missing
+	if ~isfield(x.Q, 'TimeEncodedMatrixType')
+		x.Q.TimeEncodedMatrixType = [];
+	end
+
+	if ~isfield(x.Q, 'TimeEncodedMatrixSize')
+		x.Q.TimeEncodedMatrixSize = [];
+	end
+
+	if ~isfield(x.Q, 'TimeEncodedMatrix')
+		x.Q.TimeEncodedMatrix = [];
+	end
+
+	% Check for duplicate definitions
+	if ~isempty(x.Q.TimeEncodedMatrixType) && ~isempty(x.Q.TimeEncodedMatrix)
+		error('Both TimeEncodedMatrixType defined and TimeENcodedMatrix is provided. Please provide only one parameter.')
```

Comment:
Bit weird error. The user could perfectly fine add the correct TimeEncodedMatrix that belongs to the MatrixType right.

BTW: these TimeEncoded parameters belong to the data (so to the ASL json) not to the dataPar/pipeline settings right?
And they are strictly a quantification parameter, why are they checked here?

This is a good example. Inside xASL_im_HadamardDecoding you could nicely check if TimeEncodedMatrix & MatrixType suit one another, instead of a quick and dirty check here.

Think of all the other data parameter checks that we would have to move here, to be consistent. This would be the full BIDS stuff, like checking that PASL doesn't have labeling duration. To understand the content, the same comments that are in the quantification functions need to be copied here...

As a temporary solution, what we could do, is create a separate `xASL_quant_ParameterCheck` function that checks all these parameters. This can then be called at the start of xASL_module_ASL as well as at the start of xASL_wrp_Quantify.

Reply:
yes. studyPar.json. Good idea to check if it really differs, then use the matrix itself and issue a warning. I'm doing that now in the HadamardDecode as we need the correct matrix which is obtained there. Done in 63feb017

Other than that. This is not quick and dirty solution. You wanted to have these checks there. I just added the new custom matrix.

It can't be done in Wrp_Quantify because we need to have that already while doing the subtraction.
We can't do that in quant_ParamCheck as we can only do this later when we have the correct matrix.
We can't do this as part of BIDS check as this is not a BIDS parameter.

Reply:
I don't follow. Move where? 
Some checks need to be done at the start of ASL module, some later but before Quantify, some later but before subtraction.

So where you want to call this parameterCheck function? Cause you can't call all at the same time. you have to call some right at the start, but not later. ANd other check cannot be called at the start, but only later. 
So I don't see where you could call parameter check at the same time.

But of course - all parameter checks that are at the start of the ASL module will go to a sub-function. We have already agreed upon that.